lstchain/tests/test_lstchain.py

```diff
@@ -1,24 +1,22 @@
 from ctapipe.utils import get_dataset_path
 import numpy as np
 import pytest
-import os
-import shutil
+import tempfile
 import pandas as pd
 from lstchain.io.io import dl1_params_lstcam_key, dl2_params_lstcam_key
 from lstchain.reco.utils import filter_events
-import astropy.units as u 
+import astropy.units as u
+from pathlib import Path
 
-test_dir = 'testfiles'
 
-os.makedirs(test_dir, exist_ok=True)
+mc_gamma_testfile = Path(get_dataset_path('gamma_test_large.simtel.gz'))
+
+
+@pytest.fixture(scope='session')
+def temp_dir():
+    with tempfile.TemporaryDirectory() as d:
+        yield Path(d)
 
-mc_gamma_testfile = get_dataset_path('gamma_test_large.simtel.gz')
-dl1_file = os.path.join(test_dir, 'dl1_gamma_test_large.simtel.h5')
-dl2_file = os.path.join(test_dir, 'dl2_gamma_test_large.simtel.h5')
-fake_dl2_proton_file = os.path.join(test_dir, 'dl2_fake_proton.simtel.h5')
-file_model_energy = os.path.join(test_dir, 'reg_energy.sav')
-file_model_disp = os.path.join(test_dir, 'reg_disp_vector.sav')
-file_model_gh_sep = os.path.join(test_dir, 'cls_gh.sav')
```

Comment:
Was it really an issue to have these files defined globally?
It allows other tests to import the paths and define them in a single place.

Reply:
The whole point of this was to get to a state where tests don't depend on each other, so that single tests can be run and stuff like `pytest --last-failed` works. I'll change this to a draft PR and come back as soon as I have something that works to this regard.